kernel/libs/aster-bigtcp/src/socket/bound.rs

```diff
@@ -264,50 +320,67 @@ impl Deref for NeedIfacePoll {
     }
 }
 
-impl<E: Ext> BoundTcpSocket<E> {
+impl<E: Ext> TcpConnection<E> {
     /// Connects to a remote endpoint.
     ///
     /// Polling the iface is _always_ required after this method succeeds.
-    pub fn connect(
-        &self,
+    pub fn new_connect(
+        bound: BoundPort<E>,
         remote_endpoint: IpEndpoint,
-    ) -> Result<(), smoltcp::socket::tcp::ConnectError> {
-        let common = self.iface().common();
-        let mut iface = common.interface();
+        option: &RawTcpOption,
+        observer: E::TcpEventObserver,
+    ) -> Result<Self, (BoundPort<E>, smoltcp::socket::tcp::ConnectError)> {
+        let socket = {
+            let mut socket = new_tcp_socket();
 
-        let mut socket = self.0.socket.lock();
+            option.apply(&mut socket);
 
-        socket.connect(iface.context(), remote_endpoint, self.0.port)?;
+            let common = bound.iface().common();
+            let mut iface = common.interface();
 
-        socket.has_connected = false;
-        self.0.update_next_poll_at_ms(PollAt::Now);
+            if let Err(err) = socket.connect(iface.context(), remote_endpoint, bound.port()) {
+                drop(iface);
```

Comment:
Any purpose to call `drop(iface)` explicitly here? Otherwise `iface` should also be implicitly dropped.

Reply:
Removing this `drop` will cause compiler errors:
```text
error[E0505]: cannot move out of `bound` because it is borrowed
   --> kernel/libs/aster-bigtcp/src/socket/bound.rs:342:29
    |
328 |         bound: BoundPort<E>,
    |         ----- binding `bound` declared here
...
338 |             let common = bound.iface().common();
    |                          ----- borrow of `bound` occurs here
...
342 |                 return Err((bound, err));
    |                             ^^^^^ move out of `bound` occurs here
...
346 |         };
    |         - borrow might be used here, when `iface` is dropped and runs the `Drop` code for type `ostd::sync::spin::SpinLockGuard_`
    |
```

Reply:
Thanks for your explanation!